kernelci.org/content/en/docs/tests/howto.md

```diff
@@ -1,334 +1,336 @@
-          - lava-test-case test-btrfs-cli-001 --shell ls /usr/bin/btrfs-verify-cli.sh
+          - lava-test-case uname-os --shell '[ $(uname -o) = "GNU/Linux" ]'
       from: inline
       name: {{ plan }}
       path: inline/{{ plan }}.yaml
+      lava-signal: kmsg
 ```
 
+This is pretty much all boiler plate, except for the `lava-test-case` line
+which runs the test and uses the exit code to set the result (0 is pass, 1 is
+fail).  Some extra templates need to be added for each boot method, such as
+GRUB, U-Boot and Depthcharge.  For example, here's the Depthcharge one to use
+on Chromebooks `generic-depthcharge-tftp-nfs-uname-template.jinja2`:
 
-```
-cat templates/btrfs/generic-qemu-btrfs-template.jinja2
-```
-
-```
-{% extends 'boot/generic-qemu-boot-template.jinja2' %}
+```yaml
+{% extends 'boot-nfs/generic-depthcharge-tftp-nfs-template.jinja2' %}
 {% block actions %}
 {{ super () }}
 
-{% include 'btrfs/btrfs.jinja2' %}
+{% include 'uname/uname.jinja2' %}
 
 {% endblock %}
-
-{%- block image_arg %}
-        image_arg: '-kernel {kernel} -append "console={{ console_dev }},115200 root=/dev/ram0 debug verbose {{ extra_kernel_args }}"'
-{%- endblock %}
 ```
 
-3. Update `test-plans` section of `test-configs.yaml` with new test definition
+The name of the template follows a convention to automatically generate the
+variant required for a particular platform.  This one is for the `uname` test
+plan on a platform using `depthcharge`, with the kernel downloaded over `tftp`
+and the rootfs available over `nfs`.
+
+### KernelCI YAML configuration
+
+See commit on the [how-to
+branch](https://github.com/kernelci/kernelci-core/commits/how-to): `config/core:
+enable uname test plan using Buster NFS`
+
+Once the LAVA templates have been created, the next step is to enable the test
+plan in the [KernelCI YAML configuration](../../core/config).
+
+First add the `uname` test plan with the chosen rootfs (Debian Buster NFS in
+this case) in `test-configs.yaml`:
 
+```yaml
+test_plans:
+
+  uname:
+    rootfs: debian_buster_nfs
+```
+
+Then define which platforms should run this test plan, still in
+`test-configs.yaml`:
 
 ```yaml
-  btrfs:
-    rootfs: debian_buster_btrfs_ramdisk
-    pattern: 'btrfs/generic-qemu-btrfs-template.jinja2'
+test_configs:
+
+  - device_type: hp-11A-G6-EE-grunt
+    test_plans:
+      - uname
+
+  - device_type: minnowboard-max-E3825
+    test_plans:
+      - uname
 ```
 
-4. Use `kci_build` to build the kernel and `kci_test` to run the tests in a LAVA lab.
+Each test plan also needs to be enabled to run in particular test labs in
+`lab-configs.yaml`.  Some labs such as the Collabora one allow all tests to be
+run, and it contains the platforms listed above so no extra changes are
+required at this point.
+
+
+These changes are enough to make an intial pull request in
+[`kernelci-core`](https://github.com/kernelci/kernelci-core), and the test will
+automatically get run on [staging](../../instances/staging).  Then the results
```

Comment:
Assuming that the creator is on the whitelist in [staging.ini](https://github.com/kernelci/kernelci-deploy/blob/main/data/staging.ini)?

Reply:
When someone sends a pull request, if they're genuine contributors they'll get added to the list by a maintainer.

Reply:
OK I've added a Note block about that.

Reply:
Thanks